neural_compressor/torch/quantization/algorithm_entry.py

```diff
@@ -156,6 +157,62 @@ def static_quant_entry(
     return q_model
 
 
+###################### Smooth Quant Algo Entry ##################################
+@register_algo(name=SMOOTH_QUANT)
+@torch.no_grad()
+def smooth_quant_entry(
+    model: torch.nn.Module, configs_mapping: Dict[Tuple[str, callable], SmoothQuantConfig], *args, **kwargs
+) -> torch.nn.Module:
+    logger.info("Quantize model with the smooth quant algorithm.")
+    from neural_compressor.torch.algorithms.smooth_quant.smooth_quant import smooth_quantize
+
+    # rebuild tune_cfg for smooth_quant function
```

Comment:
```suggestion
    # Convert the user config into internal format.
```
nit: Quantization should not aware tuning-related matters. This comment may confuse the user.